### PR TITLE
server: fix fail to fetch profile by pprof

### PR DESCRIPTION
**Pull request by fipped**
When we use `go tool pprof http://127.0.0.1:10080/debug/pprof/`, we will get "failed to fetch any profile" message due to last modification of start HTTP server.
This PR fixes this problem.
@coocood @tiancaiamao PTAL

### CODE REVIEW

File: server/http_status.go [hunk 1]
```diff
@@ -74,10 +74,10 @@ func (s *Server) startHTTPServer() {
 	}
 	log.Infof("Listening on %v for status and metrics report.", addr)
 
-	serverMux := http.NewServeMux()
-	serverMux.Handle("/", router)
+	http.DefaultServeMux = http.NewServeMux()
```
Comment (tiancaiamao) [inline server/http_status.go]: I guess this line can be removed

**Comment by shenli** [comment]
Please add a test case like [this](https://github.com/pingcap/tidb/blob/master/server/server_test.go#L835) to make sure pprof pages could be accessed normally.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**